components/esp_lvgl_port/esp_lvgl_port.c

```diff
@@ -0,0 +1,455 @@
+#else
+#define LVGL_PORT_HANDLE_FLUSH_READY 1
+#endif
+
+static const char *TAG = "LVGL";
+
+/*******************************************************************************
+* Types definitions
+*******************************************************************************/
+
+typedef struct lvgl_port_ctx_s {
+    SemaphoreHandle_t   lvgl_mux;
+    esp_timer_handle_t  tick_timer;
+    bool                running;
+} lvgl_port_ctx_t;
+
+typedef struct {
+    esp_lcd_panel_io_handle_t io_handle;    /* LCD panel IO handle */
+    esp_lcd_panel_handle_t    panel_handle; /* LCD panel handle */
+    lvgl_port_rotation_cfg_t  rotation;     /* Default values of the screen rotation */
+    lv_disp_drv_t             disp_drv;     /* LVGL display driver */
+} lvgl_port_display_ctx_t;
+
+typedef struct {
+    esp_lcd_touch_handle_t   handle;     /* LCD touch IO handle */
+    lv_indev_drv_t           indev_drv;  /* LVGL input device driver */
+} lvgl_port_touch_ctx_t;
+
+/*******************************************************************************
+* Local variables
+*******************************************************************************/
+static lvgl_port_ctx_t lvgl_port_ctx;
+static int lvgl_port_timer_period_ms = 5;
+
+/*******************************************************************************
+* Function definitions
+*******************************************************************************/
+static void lvgl_port_task(void *arg);
+static esp_err_t lvgl_port_tick_init(void);
+static void lvgl_port_task_deinit(void);
+
+// LVGL callbacks
+#if LVGL_PORT_HANDLE_FLUSH_READY
+static bool lvgl_port_flush_ready_callback(esp_lcd_panel_io_handle_t panel_io, esp_lcd_panel_io_event_data_t *edata, void *user_ctx);
+#endif
+static void lvgl_port_flush_callback(lv_disp_drv_t *drv, const lv_area_t *area, lv_color_t *color_map);
+static void lvgl_port_update_callback(lv_disp_drv_t *drv);
+static void lvgl_port_touchpad_read(lv_indev_drv_t *indev_drv, lv_indev_data_t *data);
+static void lvgl_port_pix_monochrome_callback(lv_disp_drv_t *drv, uint8_t *buf, lv_coord_t buf_w, lv_coord_t x, lv_coord_t y, lv_color_t color, lv_opa_t opa);
+/*******************************************************************************
+* Public API functions
+*******************************************************************************/
+
+esp_err_t lvgl_port_init(const lvgl_port_cfg_t *cfg)
+{
+    esp_err_t ret = ESP_OK;
+    memset(&lvgl_port_ctx, 0, sizeof(lvgl_port_ctx));
+
+    assert(cfg);
+
+    /* LVGL init */
+    lv_init();
+    /* Tick init */
+    lvgl_port_timer_period_ms = cfg->timer_period_ms;
+    ESP_RETURN_ON_ERROR(lvgl_port_tick_init(), TAG, "");
+    /* Create task */
+    lvgl_port_ctx.lvgl_mux = xSemaphoreCreateMutex();
+    ESP_GOTO_ON_FALSE(lvgl_port_ctx.lvgl_mux, ESP_ERR_NO_MEM, err, TAG, "Create LVGL mutex fail!");
+    BaseType_t res = xTaskCreate(lvgl_port_task, "LVGL task", cfg->task_stack, NULL, cfg->task_priority, NULL);
```

Comment:
Can a config option be added for pinning this task to a specific core (default can be floating)?